### ISSUE TITLE
"After a reboot, this app can't start until you unlock your phone"

### ISSUE THREAD

**Issue by wrsg** [klausw/hackerskeyboard#587]
Android 7.1.1, Nexus 6. 
Other non-Google keyboards don't display this warning so I'm assuming they are authenticated by the system at the point when they are enabled/installed. I tested SwiftKey and it's fine. 
![screenshot_20170707-084529](https://user-images.githubusercontent.com/44514/27948267-289278d0-62f1-11e7-9ff3-8f27ab2ff1b3.png)

**Comment by TPS** [comment]
This must be a Nexus ROM thing; how do _you_ get that message? _I_ use HK on 7.1.1 initial launcher lockscreen after boot, though ROM keyboard is required for initial SIM/phone unlock.

**Comment by wrsg** [comment]
I simply installed HK and then went to that keyboard enablement page in Settings and attempted to enable HK, then that pops up. It's working fine, that message is just saying the phone won't allow me to unlock at startup using HK.

**Comment by TPS** [comment]
I just confirmed it.  I upgraded my ROM from 7.0 _after_ installing/enabling HK, so I'd never seen it.  By chance, is Swiftkey _part_ of your ROM _or_ installed System?  That'd probably bypass the message.  Do you have any other keyboards _not_ installed System that _don't_ show the message?  All mine do, _except_ GBoard, but that was part of my ROM.

**Comment by wrsg** [comment]
No, I just downloaded Switfkey (Beta) to see if it was a good replacement for the system keyboard (which kept crashing). No other keyboards that I use. Tbh the AOSP keyboard would be fine but I can't find it now.

**Comment by TPS** [comment]
Fascinating.  I've replicated all of the above w/ Swiftkey ß.  Maybe it's coded w/ compatibility for [Direct Boot](https://developer.android.com/training/articles/direct-boot.html)?  If so, that'd be a great thing for HK, though it'd probably mean rewriting targeting Nougat, 5 Android versions _ahead_ of where it's now!

_OT: If you've Nexus stock ROM, AOSP keyboard __should__ be your native keyboard (though now it's called [GBoard](http://play.google.com/store/apps/details?id=com.google.android.inputmethod.latin).)_

**Comment by Agaff** [comment]
www.fxsitecompat.com/en-CA/docs/2017/insecure-login-forms-now-disable-autofill-show-warning-beneath-input-control/

**Comment by TPS** [comment]
> _www.fxsitecompat.com/en-CA/docs/2017/insecure-login-forms-now-disable-autofill-show-warning-beneath-input-control/_

_@Agaff Completely OT to the point of non sequitur, but also an excellent & welcome change by Mozilla._

**Comment by A-Jarvis-Studio** [comment]
> I had this problem and the way to solve it was to enable and set the keyboard via the command line.
> Steps:
> 
> 1. Go to shell
> 2. $ pm list packages
> 3. $ ime enable "your_preferred_package"
> 4. $ ime list -a|grep mId
> 5. $ ime set "the_mId_of_your_keyboard"
> 
> Hope it helps

Sorry Can U Tell Me How to do it at the Android Studio 
or Where can I do for it

**Comment by manos-saratsis** [comment]
You need to do it via the Android Debug Bridge ([adb](https://developer.android.com/studio/command-line/adb)). I do it from the terminal but you can even set it up from within Android studio, there are instructions on the web.
